src/components/CustomMessage.tsx

```diff
@@ -45,6 +47,16 @@ export default function CustomMessage(props: Props) {
   const firstMessage: UserMessage = allMessages[0] as UserMessage;
   const firstMessageId = firstMessage?.messageId ?? -1;
 
+  const isBotWelcomeMessage = useMemo(() => {
+    const botWelcomeMessages = getBotWelcomeMessages(
+      allMessages,
+      botUser.userId
+    );
+    return !!botWelcomeMessages.find(
+      (welcomeMessage) => welcomeMessage.messageId === message.messageId
+    );
+  }, [allMessages.length]);
+
```

Comment:
How about this instead of searching for all the messages?

```suggestion
  const isBotWelcomeMessage = message.sender?.userId === botUser.userId // isBotMessage(message, botUser);
```

Reply:
Oh I was trying to get only the welcoming messages not the all bot messages.
![Screenshot 2023-09-12 at 5 21 22 PM](https://github.com/sendbird/chat-ai-widget/assets/10060731/ed63ea65-d682-4fe0-9002-4a1b61fe3f2f)

So these are the welcome messages.

Reply:
Ahh,, I understand why it's a "welcome message" ㅋㅋㅋ

How about lifting it up to a higher level rather than recalculating it in each message component?

Reply:
Yeah sounds good. https://github.com/sendbird/chat-ai-widget/pull/56/commits/ca4fa47fa9f78510df395d9b2301c7fffdfbb67a Just addressed it :)